core/internal/flux_monitor_simulated_blockchain_test.go

```diff
@@ -0,0 +1,370 @@
+	// - deploy a brand new FM contract
+	var description [32]byte
+	copy(description[:], "exactly thirty-two characters!!!")
+	fa := deployFluxAggregator(t, fee, 1, 8, description)
+
+	// Set up chainlink app
+	config, cfgCleanup := cltest.NewConfig(t)
+	config.Config.Set("DEFAULT_HTTP_TIMEOUT", "100ms")
+	timeout := 100 * time.Millisecond // if failing due to timeouts, increase this
+	defer cfgCleanup()
+	app, cleanup := cltest.NewApplicationWithConfigAndKeyOnSimulatedBlockchain(t,
+		config, fa.backend)
+	defer cleanup()
+	require.NoError(t, app.StartAndConnect(), "failed to start chainlink")
+	minFee := app.Store.Config.MinimumContractPayment().ToInt().Int64()
+	require.Equal(t, fee, minFee, "fee paid by FluxAggregator (%d) must at "+
+		"least match MinimumContractPayment (%s). (Which is currently set in "+
+		"cltest.go.)", fee, minFee)
+
+	answer := int64(1) // Answer the nodes give on the first round
+
+	//- have one of the fake nodes start a round.
+	roundId := int64(1)
+	processedAnswer := answer * 100 /* job has multiply times 100 */
+	updateAnswer(t, answerParams{fa: &fa, roundId: roundId,
+		answer: processedAnswer, from: fa.neil, isNewRound: true,
+		completesAnswer: false})
+
+	// - successfully close the round through the submissions of the other nodes
+	// Response by malicious chainlink node, nallory
+	initialBalance := currentBalance(t, &fa).Int64()
+	reportPrice := answer
+	priceResponse := func() string {
+		return fmt.Sprintf(`{"data":{"result": %d}}`, reportPrice)
+	}
+	mockServer := cltest.NewHTTPMockServerWithAlterableResponse(t, priceResponse)
+	defer mockServer.Close()
+
+	// When event appears on submissionReceived, flux monitor job run is complete
+	submissionReceived := make(chan *faw.FluxAggregatorSubmissionReceived)
+	subscription, err := fa.aggregatorContract.WatchSubmissionReceived(nil,
+		submissionReceived, []*big.Int{}, []uint32{},
+		[]common.Address{fa.nallory.From})
+	require.NoError(t, err, "failed to subscribe to SubmissionReceived events")
+	defer subscription.Unsubscribe()
+
+	// Create FM Job, and wait for job run to start (the above UpdateAnswer call
+	// to FluxAggregator contract initiates a run.)
+	buffer := cltest.MustReadFile(t, "testdata/flux_monitor_job.json")
+	var job models.JobSpec
+	require.NoError(t, json.Unmarshal(buffer, &job))
+	initr := &job.Initiators[0]
+	initr.InitiatorParams.Feeds = cltest.JSONFromString(t, fmt.Sprintf(`["%s"]`,
+		mockServer.URL))
+	initr.InitiatorParams.PollingInterval =
+		models.Duration(timeout)
+	initr.InitiatorParams.Address = fa.aggregatorContractAddress
+	j := cltest.CreateJobSpecViaWeb(t, app, job)
+	jrs := cltest.WaitForRuns(t, j, app.Store, 1) // Submit answer from
+	reportedPrice := jrs[0].RunRequest.RequestParams.Get("result").String()
+	assert.Equal(t, reportedPrice, fmt.Sprintf("%d", reportPrice),
+		"failed to report correct price to contract")
+	var receiptBlock uint64
+	select { // block until FluxAggregator contract acknowledges chainlink message
```

Comment:
I tried this, and because I would still need a `switch` statement to get a non-blocking read in the argument to `gomega.Eventually`, I decided it's clearer the way it is. Please LMK if I've missed something.

Reply:
I don't really understand `t.Helper`'s purpose. Why would I want to elide information from test failure output?